### DISCUSSION TITLE
No output files defined for this Tool specification error

### DISCUSSION THREAD

**Discussion by Henryutd2021** [spine-tools/SpineOpt.jl#614] [Support]
Hello everyone,

I followed the steps listed on https://spine-toolbox.readthedocs.io/en/latest/simple_system.html strictly to create my practice project, but every time I run it, I get the following error, I'm a beginner, so I haven't solved the problem despite trying many things. Could you please help me with the problem and if you can provide further solutions would be appreciated.

![image](https://user-images.githubusercontent.com/89668896/231778644-70aa704c-b50c-4c3f-86cb-ac4ef621faea.png)

**Comment by jkiviluo** [comment]
@Henryutd2021 has created a result database, so that shouldn't be it. Could someone help him out? @Tasqu?

**Comment by DillonJ** [comment]
Hi @Henryutd2021,

The message about there being no output file specified is just a warning and is not the cause of the error. Are you seeing any errors in the tool's console? What you have shown above is the general workflow log. When tools run, they usually have their own console window which usually shows more detailed progress.

See below. You need to click on the tool and then click on the execution then you should see the detailed output of the tool.

![Capture](https://user-images.githubusercontent.com/7080191/232711973-3bd6f9ef-e7db-47ce-b983-03b8b06571d5.JPG)

**Comment by LouisFliche** [comment]
Hello @Henryutd2021, 

I had a similar problem recently. If it is the same one, it seems to be a problem with the version of SpineInterface. Most likely you don't have the most recent one, and this creates problems in running SpineOpt. 
Could you check with Pkg.status() what version of SpineInterface you have? It should normally be 0.10.3.

If you indeed don't have the latest version of SpineInterface, it is possible that trying to update it with pkg"update SpineInterface" doesn't work because of depency restrictions with SpineOpt (at least if it the same problem as I had). 
So, I suggest you update your version of SpineInterface by pulling the git repo directly. Then, check that you now have the latest version of SpineInterface in your Julia environnement, and try rerunning your system (with a new Julia kernel). Hope this works!

If it still doesn't work or if you already had the latest version of SpineInterface, could you please send a screenshot of your Julia console for SpineOpt (click on the SpineOpt tool in the Design View, and then on object_activity_control in the console window) - and also possibly your version number of the different Spine tools (Pkg.status())?

**Reply by jkiviluo** [reply]
Getting the latest SpineInterface from GitHub instead of using the version from the Julia registry can be done with these commands in the Julia REPL (can be done in the Toolbox Julia console too).

```
using Pkg
Pkg.add(url="https://github.com/spine-tools/SpineInterface.jl.git")
```

Once added from GitHub, it can then be updated from GitHub using

```
Pkg.update("SpineInterface")
```

**Comment by Henryutd2021** [comment]
Thank you @jkiviluo, @LouisFliche, and @DillonJ. I re-installed  spinetoolbox from sources using Git. I followed @LouisFliche method to check the version of spineinterface, and it was not the latest one, and then， I updated the spineinterface using the method @jkiviluo  mentioned. It eventually ran successfully. I also learned how to view console information from @DillonJ mentioned method.Thank you again for your help.
